File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        return duplicate(nums);
    }
    public static boolean duplicate(int[] arr) {
        Arrays.sort(arr); 
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] == arr[i - 1]) {
                return true; 
            }
        }
        return false;
    }

}

File: 217-contains-duplicate/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> st;
        for(int i=0; i<nums.size(); i++){
            if(!st.insert(nums[i]).second){
                return true;
            }
        }
        return false;

        //st.insert(num).second == true → element was inserted (not a duplicate).
       // st.insert(num).second == false → element already exists (duplicate found).
        
        
    }
};